internal/admission/validator.go

```diff
@@ -304,6 +306,59 @@ func (validator KongHTTPValidator) ValidateGateway(
 	return true, "", nil
 }
 
+func (validator KongHTTPValidator) ValidateHTTPRoute(
+	ctx context.Context, httproute gatewayv1alpha2.HTTPRoute,
+) (bool, string, error) {
+	if len(httproute.Spec.ParentRefs) == 0 {
+		// we don't try to validate an HTTPRoute if it doesn't have parentRefs
+		// configured, because there's no way we can be sure the route is managed
+		// by this controller implementation.
+		return true, "", nil
+	}
+
+	// in order to be sure whether or not an HTTPRoute resource is managed by this
+	// controller we disallow references to Gateway resources that do not exist.
+	var managedGateways []*gatewayv1alpha2.Gateway
+	for _, parentRef := range httproute.Spec.ParentRefs {
```

Comment:
duplication of class matching logic defined in https://github.com/Kong/kubernetes-ingress-controller/blob/main/internal/controllers/gateway/gateway_utils.go

Reply:
Maybe I'm missing what function you're talking about, or misunderstanding in some other way but I'm not seeing the duplication?

Reply:
discussed in zoom: @mflendrich will make a follow up PR fixing the abstraction here